### ISSUE TITLE
Ability to create an empty table with specified types in the Table test suite

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#6159]
Currently, the architecture of `table_builder` used in `Common_Table_Operations` part of our Table test suite relies on building an in-memory `Table` and then uploading it to a Database (or returning as-is for the in-memory backend).

When we try to build a table with 0-rows, that does not allow our `Table.new` to infer the type of each column (no way to infer a type if no data is provided), so the columns end up as `Mixed` which is not supported by most databases.

Some tests rely on tables with no rows.

Currently, we construct a non-empty table and filter it out to make it empty.

Ideally, we should have some extension to `table_builder` allowing to specify the `Value_Type` explicitly and create 0-row tables with specified types without such workarounds.

**Comment by radeusgd** [comment]
I think very possibly that the simplest solution may be to just keep creating a column of `Nothing`s that will end up being `Mixed` and just use `cast` (#6112) to change it to the desired type. Then we do not need to heavily alter how the test suite is built.

We could then just do:
```
t0 = table_builder [["X", [Nothing]]]
t = t0.cast "X" Value_Type.Integer
```

**Comment by radeusgd** [comment]
I think this will be possible to be implemented quite easily once we do #6111 - with it we will be able to explicitly specify the Value_Type on `Column.from_vector`.
We can then amend the `table_builder` to just take an in-memory Table and convert it to the desired backend (so for in-memory tests it will be identity and for database tests it will upload that table). This should be the simplest way to get it working without adding some excess APIs and just using what we have nicely, and in a quite uniform way.

Alternativel from `Column.from_vector`, one can create an in-memory Table with null columns and `cast` these columns to the desired type. (This still aligns with changing `table_builder` to take an in-memory table instead of column descriptions).

**Comment by jdunkerley** [comment]
Closing as current workaround is sufficient for now.
